command/agent/config_test.go

```diff
@@ -1068,3 +1071,109 @@ func TestReadConfigPaths_dir(t *testing.T) {
 		t.Fatalf("bad: %#v", config)
 	}
 }
+
+func TestUnixSockets(t *testing.T) {
```

Comment:
This test is fairly big and has a pretty broad scope. Maybe we can instead have a helper function to which you would pass a test function and some data? What do you think?